src/wallet/api/wallet.cpp

```diff
@@ -1994,12 +2010,16 @@ void WalletImpl::refreshThreadFunc()
 
 void WalletImpl::doRefresh()
 {
+    bool rescan = m_refreshShouldRescan.exchange(false);
     // synchronizing async and sync refresh calls
     boost::lock_guard<boost::mutex> guarg(m_refreshMutex2);
-    try {
+    do try {
+        LOG_PRINT_L3(__FUNCTION__ << ": doRefresh, rescan = "<<rescan);
         // Syncing daemon and refreshing wallet simultaneously is very resource intensive.
         // Disable refresh if wallet is disconnected or daemon isn't synced.
         if (m_wallet->light_wallet() || daemonSynced()) {
+            if(rescan)
+                m_wallet->rescan_blockchain(false);
```

Comment:
I doubt it works with light wallet.

Reply:
I didn't see any specific `rescan_bc` implementation for light wallet. I just copied implementation of `rescan_bc` into WalletAPI. Could you please elaborate on this?

Reply:
rescan_bc scans the blockchain from the daemon. A light wallet does not have a (normal) daemon to talk to.